easybuild/easyblocks/m/mxnet.py

```diff
@@ -0,0 +1,222 @@
+        """Specify list of files to copy"""
+        self.cfg['files_to_copy'] = ['bin', 'include', 'lib',
+                                     (['dmlc-core/include/dmlc', 'nnvm/include/nnvm'], 'include')]
+        super(EB_MXNet, self).install_step()
+
+    def extensions_step(self):
+        """Build & Install both Python and R extension"""
+        # we start with the python bindings
+        self.py_ext.src = os.path.join(self.mxnet_src_dir, "python")
+        change_dir(self.py_ext.src)
+
+        self.py_ext.prerun()
+        self.py_ext.run(unpack_src=False)
+        self.py_ext.postrun()
+
+        # next up, the R bindings
+        self.r_ext.src = os.path.join(self.mxnet_src_dir, "R-package")
+        change_dir(self.r_ext.src)
+        mkdir("inst")
+        symlink(os.path.join(self.installdir, "lib"), os.path.join("inst", "libs"))
+        symlink(os.path.join(self.installdir, "include"), os.path.join("inst", "include"))
+
+        # MXNet doesn't provide a list of its R dependencies by default
+        namespace = """# Export all names
+exportPattern(".")
+
+# Import all packages listed as Imports or Depends
+import(
+methods,
+Rcpp,
+DiagrammeR,
+data.table,
+jsonlite,
+magrittr,
+stringr
+)
+"""
+        write_file("NAMESPACE", namespace)
+        change_dir(self.mxnet_src_dir)
+        self.r_ext.prerun()
+        # MXNet is just weird. To install the R extension, we have to:
+        # - First install the extension like it is
+        # - Let R export the extension again. By doing this, all the dependencies get
+        #   correctly filled and some mappings are done
+        # - Reinstal the exported version
+        self.r_ext.run()
+        run_cmd("R_LIBS=%s Rscript -e \"require(mxnet); mxnet:::mxnet.export(\\\"R-package\\\")\"" % self.installdir)
+        change_dir(self.r_ext.src)
+        self.r_ext.run()
+        self.r_ext.postrun()
+
+    def sanity_check_step(self):
+        """Check for main library files for MXNet"""
+        # for the extension we are doing the loading of the fake module ourself
+        try:
+            fake_mod_data = self.load_fake_module()
+        except EasyBuildError, err:
+            raise EasyBuildError("Loading fake module failed: %s", err)
+
+        if not self.py_ext.sanity_check_step():
+            raise EasyBuildError("The sanity check of the Python bindings failed")
+
+        if not self.r_ext.sanity_check_step():
+            raise EasyBuildError("The sanity check of the R bindings failed")
+
+        self.clean_up_fake_module(fake_mod_data)
+
+        custom_paths = {
+            'files': ["lib/libmxnet.%s" % ext for ext in ['a', get_shared_lib_ext()]],
+            'dirs': [],
+        }
+        super(EB_MXNet, self).sanity_check_step(custom_paths=custom_paths)
+
+    def make_module_extra(self, *args, **kwargs):
+        """Custom variables for MXNet module."""
+        txt = super(EB_MXNet, self).make_module_extra(*args, **kwargs)
+
+        self.py_ext.set_pylibdirs()
```

Comment:
hmm, do this right in `prepare_step`, since it's now being done twice now (when fake module is created, and again when real module is created)